uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/ContactTest.php

```diff
@@ -190,4 +187,78 @@ public function testGetLeaveManageesReturnsEmptyResultsWhenStaffMemberIsTryingTo
 
     $this->assertEquals(0, $result['count']);
   }
+
+  /**
+   * @expectedExceptionMessage Either unassigned must be true or managed_by parameter present
+   * @expectedException CiviCRM_API3_Exception
+   */
+  public function testGetLeaveManageesThrowsAnExceptionWhenUnassignedIsFalseAndManagedByIsAbsent() {
+    $this->callAPI('Contact', 'getleavemanagees', ['unassigned' => false]);
+  }
+
+  /**
+   * @expectedExceptionMessage Unassigned cannot be true and managed_by parameter also present
+   * @expectedException CiviCRM_API3_Exception
+   */
+  public function testGetLeaveManageesThrowsAnExceptionWhenUnassignedIsTrueAndManagedByParameterIsPresent() {
+    $this->callAPI('Contact', 'getleavemanagees', ['unassigned' => true, 'managed_by' => 1]);
+  }
+
+  /**
+   * @expectedExceptionMessage Either unassigned must be true or managed_by parameter present
+   * @expectedException CiviCRM_API3_Exception
+   */
+  public function testGetLeaveManageesThrowsAnExceptionWhenUnassignedAndManagedByParameterIsAbsent() {
+    $this->callAPI('Contact', 'getleavemanagees', []);
+  }
+
+  public function testGetLeaveManageesReturnsOnlyContactsWithoutActiveLeaveApproverRelationshipWhenUnassignedTrue() {
+    $contact1 = $this->contact1;
+    $contact2 = $this->contact2;
+    $contact3 = ContactFabricator::fabricate();
+    $manager = $this->manager;
+
+    $relationshipType = RelationshipTypeFabricator::fabricate();
```

Comment:
Please add a comment here noting why $contact3 won't be returned, even though this contact has a relationship with a manager